### ISSUE TITLE
feat: Assign labels a color

### ISSUE THREAD

**Issue by adamslc** [git-bug/git-bug#59]
It would be nice to be able to assign each label a color, and use that color every time the label is printed.

**Comment by MichaelMure** [comment]
My thought on this:

- the good end game way to have properly defined colors for each label would be to have a git-bug config where admins could define various properties including what label exist and their colors, define what status exist for a bug and how you can transition from one status to another ... but we are not there yet
- there is already in the webUI some code to assign arbitrary colors to a label (https://github.com/MichaelMure/git-bug/blob/master/webui/src/Label.js#L10-L43)

My proposal:
1. migrate the webui color code into the core. That way, a label will have the same color in all UIs, will get simpler and the core APIs will be ready for the end game
2. when we have the code to define the config, we will just have to plug that in the core and we will have either automatic label colors or configured labels if users care enough to define them.

**Comment by j-f1** [comment]
For the termui, there needs to be a fallback for the colors since some terminals only support 16 or 256 colors.

**Comment by MichaelMure** [comment]
@j-f1 good point. That should be fairly easy to reduce the color precision to 16 or 8bits. I'm a bit confused as to how one is supposed to detect the terminal capabilities. I have the same question for how to detect the terminal size btw.

**Comment by MichaelMure** [comment]
@j-f1 Thanks

More info on the subject: https://gist.github.com/XVilka/8346728

I found a go implementation: https://github.com/xo/terminfo/blob/master/color.go#L49-L91

**Comment by MichaelMure** [comment]
Label's color are now defined directly in the core and used in the webUI through the graphQL API.

Next steps are:
- display the label's color in the termui
- make the colors configurable

**Comment by MichaelMure** [comment]
Code used by Github's hub command to display labels: https://github.com/github/hub/blob/master/utils/color.go

![image](https://user-images.githubusercontent.com/294669/60720763-0f904000-9f2c-11e9-8c9b-d0a5b782bef7.png)

**Comment by kondanta** [comment]
Mind if I ask the current state of this issue?

**Comment by MichaelMure** [comment]
Labels should be displayed everywhere with the correct color, but you can't choose the color yourself. That would require a proper configuration data structured stored in git and shared.

**Comment by kondanta** [comment]
Alright, I understand. Do you planning to support colour selection in near future?

**Comment by MichaelMure** [comment]
Probably not as there is some significant development with higher priority (some being required to implement that feature).

**Comment by vtexier** [comment]
My git-bug  choose the same color for two labels on a total of three labels!

![Capture du 2020-11-12 17-10-42](https://user-images.githubusercontent.com/2174016/98966298-65ef9480-250b-11eb-972d-23f9c1ce4108.png)

A silly idea: replace or add to colors a special character symbol.

Color blind people, and others could distinguish issues more easily...

**Comment by dllud** [comment]
I second @vtexier. On the CLI and TUI, displaying a (coloured) character (e.g. the first letter of the label) would be much more useful than the current square.

I just created 2 labels on a repo and git-bug chose exactly the same colour for them.

**Comment by dllud** [comment]
Well, spamming the issue then… 
It's still relevant to me. I wouldn't like it to be closed by a bot.

(I find this kind of bot detrimental. What's the problem with keeping unfixed issues open?…)

**Comment by sudoforge** [comment]
> I find this kind of bot detrimental. What's the problem with keeping unfixed issues open?

:wave: hey there, i added the stale bot workflow. it's since been refactored so that it [will never close issues or pull requests](https://github.com/michaelmure/git-bug/pull/1224).

it is primarily useful for larger backlogs as a way to indicate where some attention is needed, e.g. what has gone unattended for a long time and should be looked at.
